internal/firebase/verify_email_code.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package firebase is common logic and handling around firebase.
+package firebase
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type verifyEmailRequest struct {
+	Code string `json:"oobCode"`
+}
+
+// VerifyEmailCode sends a password reset email to the user. If the new
+// password is given, it applies the password reset change with the new password
+// using the code.
+//
+// See: https://firebase.google.com/docs/reference/rest/auth#section-send-password-reset-email
+func (c *Client) VerifyEmailCode(ctx context.Context, code, key string) error {
+	r := &verifyEmailRequest{Code: code}
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(r); err != nil {
+		return fmt.Errorf("failed to create json body: %w", err)
+	}
+
+	u := c.buildURL("/v1/accounts:update")
```

Comment:
Is this correct? Should it be `/v1/accounts:resetPassword` per https://firebase.google.com/docs/reference/rest/auth#section-verify-password-reset-code ?

Reply:
Unfortunately no: https://cloud.google.com/identity-platform/docs/use-rest-api#section-confirm-email-verification

Reply:
`<rant>`I hate anchor links.`</rant>`

In that case, I don't see any obvious reason why this wouldn't work?